cdap-api/src/test/java/co/cask/cdap/api/common/BytesTest.java

```diff
@@ -26,6 +31,574 @@
  */
 public class BytesTest {
 
+    @Rule public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testAdd(){
+        final byte[] a = {1, 2, 3, 4};
+        final byte[] b = {5, 6, 7, 8};
+        final byte[] empty = {};
+
+        Assert.assertArrayEquals(new byte[] {1,2,3,4,5,6,7,8}, Bytes.add(a, b));
+        Assert.assertArrayEquals(a, Bytes.add(a, empty));
+        Assert.assertArrayEquals(empty, Bytes.add(empty, empty));
+    }
+
+    @Test
+    public void testCompareTo() {
+        final byte[] a = {1, 2, 3, 4};
+        final byte[] b = {5, 6, 7, 8};
+        final byte[] empty = {};
+
+        Assert.assertEquals(0, Bytes.compareTo(empty, empty));
+        Assert.assertEquals(4, Bytes.compareTo(a, empty));
```

Comment:
The contract of `compareTo` is to return '< 0', '0' or '> 0'. Please test on the contract instead of the exact value returned.